### DISCUSSION TITLE
Are lnrpcpaymentfailurereason errors permanent?

### DISCUSSION THREAD

**Discussion by madeken** [lightningnetwork/lnd#6364] [Q&A] [answered]
When programatically making a payment, sometimes there is an error. i.e. https://api.lightning.community/#lnrpcpaymentfailurereason

However is it ever possible that one of these errors is temporary? And then it turns out the payment actually succeeded?  Or when I have one of these errors, it's final. And I know it's impossible for the payment to succeed? 

Like does an error of `FAILURE_REASON_TIMEOUT` guarantee the payment is fully timed out? Or later could find out it succeeded?

What does it look like if a payment is stuck in a limbo state, where it's not sure if the payment succeeded or failed yet?

**Comment by yyforyongyu** [comment, marked as answer]
All of the errors are final, and the payment is failed. If somehow you got one of these errors but it later succeeded, it would be a bug. In that case, submitting an issue with relevant logs will be much appreciated!

 > What does it look like if a payment is stuck in a limbo state, where it's not sure if the payment succeeded or failed yet?

In this case, your payment should have a status saying it's in-flight.
